core/src/main/resources/defaults/blueprints/cdp-data-engineering-710.bp

```diff
@@ -191,6 +210,14 @@
           {
             "name": "tez_container_size",
             "value": "4096"
+          },
+          {
+            "name": "tez_auto_reducer_parallelism",
+            "value": "false"
+          },
+          {
+            "name": "hive_service_config_safety_valve",
+            "value": "<property><name>hive.hook.proto.file.per.event</name><value>true</value></property><property><name>hive.server2.tez.sessions.per.default.queue</name><value>4</value></property>"
```

Comment:
Overriding safety-valves in the blueprints is generally a hack and does not work in all cases. Please take a look at HiveKnoxConfigProvider and make sure you are not overriding: `String principal = ConfigUtils.getSafetyValveProperty("hive.server2.authentication.spnego.principal", "HTTP/_HOST@" + realm);` in your safety valve.

Reply:
@harishjp I understand overriding safety-valves in blueprints has issues. If one makes a config change for a safety-valve in the blueprint then all other configs inside safety-valve go away (which are generally set via java code).
We wanted to change only "hive.server2.tez.sessions.per.default.queue" which was not exposed directly. But when looked at "hive_service_config_safety_valve" in CM UI, found that only "hive.hook.proto.file.per.event=true" is being set. To not remove this while modifying blueprint I have added both the configs in blueprint.

In the meantime I have already filed a [request to expose ](https://jira.cloudera.com/browse/OPSAPS-55598) "hive.server2.tez.sessions.per.default.queue" directly in CM. When this is fixed we will remove this safety-valve from blueprint.

Also, I have verified from one of my cluster that we are not overriding/removing "hive.server2.authentication.spnego.principal" while launching cluster through updated blueprint. Its values before and after modification was hive.server2.authentication.spnego.principal=HTTP/_HOST@CLOUD-PE.XCU2-8Y8X.DEV.CLDR.WORK

Reply:
The addition if hive.hook.proto.file.per.event may not be necessary here after https://jira.cloudera.com/browse/OPSAPS-54706, which was committed to 2.20.

Reply:
Have removed hive.hook.proto.file.per.event from the 4 templates now.